pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -142,12 +142,29 @@ func newXdsConnection(peerAddr string, stream DiscoveryStream) *XdsConnection {
 	}
 }
 
+// isExpectedGRPCError checks a gRPC error code and determines whether it is an expected error when
+// things are operating normally. This is basically capturing when the client disconnects.
+func isExpectedGRPCError(err error) bool {
+	if err == io.EOF {
+		return true
+	}
+
+	s := status.Convert(err)
+	if s.Code() == codes.Canceled {
+		return true
+	}
+	if s.Message() == "client disconnected" {
```

Comment:
This is really hacky, it takes me a lot time to understand, would you add some context for this?